cdk-infra/lib/constructs/worker/worker-construct.ts

```diff
@@ -0,0 +1,113 @@
+import { GatewayVpcEndpointAwsService, InterfaceVpcEndpointAwsService, Vpc } from 'aws-cdk-lib/aws-ec2';
+import {
+  AssetImageProps,
+  Cluster,
+  ContainerImage,
+  FargateService,
+  FargateTaskDefinition,
+  LogDrivers,
+} from 'aws-cdk-lib/aws-ecs';
+import { Effect, IRole, PolicyStatement, Role, ServicePrincipal } from 'aws-cdk-lib/aws-iam';
+import { LogGroup } from 'aws-cdk-lib/aws-logs';
+import { IQueue } from 'aws-cdk-lib/aws-sqs';
+import { Construct } from 'constructs';
+import path from 'path';
+
+interface WorkerConstructProps {
+  environment: Record<string, string>;
+  jobsQueue: IQueue;
+  jobUpdatesQueue: IQueue;
+}
+export class WorkerConstruct extends Construct {
+  readonly ecsTaskRole: IRole;
+  readonly taskDefinitionArn: string;
+  constructor(scope: Construct, id: string, { environment, jobsQueue, jobUpdatesQueue }: WorkerConstructProps) {
+    super(scope, id);
+
+    const isEnhanced = !!this.node.tryGetContext('enhanced');
+
+    const vpc = new Vpc(this, 'vpc', {
+      gatewayEndpoints: {
+        S3: { service: GatewayVpcEndpointAwsService.S3 },
+      },
+    });
+
+    vpc.addInterfaceEndpoint('EcrDockerEndpoint', {
+      service: InterfaceVpcEndpointAwsService.ECR_DOCKER,
+    });
+
+    vpc.addInterfaceEndpoint('EcrApiEndpoint', {
+      service: InterfaceVpcEndpointAwsService.ECR,
```

Comment:
This isn't a super important point, but I find it interesting and worth sharing nonetheless; these VPC endpoints actually cut back on our current biggest cost with our current AWS infrastructure. 

Basically, it allows us to pull from ECR through a private connection as opposed to going out to the public internet. Since the images are so large, and the costs for going from private to public internet so high, most of what we pay each month is on this.

Also, this change (at least with CDK) was really simple, and the behavior is exactly the same in terms of how we pull from ECR.